Fix running `ng serve` (alt): Set allowed hosts via environment variable

myabc opened the pull request:
**⚠️ ~~Developers will need to manually updated their `docker-compose.override.yml` on merging this PR~~** edit: not anymore

Vite now supports specifying allowed hosts as an `__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS`.

Alternative fix to #19687 / a1555af4773510da7644f1ea1a3e5357d42853a2: Only overrides allowed hosts in the development environments that need it (i.e. Docker + TLS stack).

See: https://vite.dev/config/server-options.html#server-allowedhosts
See: vitejs/vite#19325

myabc committed:
Set allowed hosts via environment variable

Vite now supports specifying allowed hosts as an
`__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS`.

Alternative fix to a1555af4773510da7644f1ea1a3e5357d42853a2: Only
overrides allowed hosts in the development environments that need it
(i.e. Docker + TLS stack).

See: https://vite.dev/config/server-options.html#server-allowedhosts
See: vitejs/vite#19325

myabc committed:
Add ports to example docker-compose.override.yml

myabc committed:
Fix typo in docker dev env README.md

[ci skip]

NobodysNightmare commented on docker/dev/tls/docker-compose.core-override.example.yml:
```diff
@@ -57,8 +59,12 @@ services:
 
   frontend:
+    environment:
+      __VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS: "openproject-assets.local"
```
Assuming that it's non-breaking for other users, could we specify this on the `docker-compose.yml` already? Then it would work out of the box for everyone using the default hostnames. If you want to use a different hostname, the override is still possible.

ba1ash committed:
Move ports and env variable to docker-compose.yml

It seems to be fine to have these ports settings as well as
__VITE_ADDITIONAL_SERVER_ALLOWED_HOSTS envvar set in docker-compose.yml
directly. So, there is less to override to get your docker TLS dev stack working.
